app/controllers/self_service/firms_controller.rb

```diff
@@ -7,5 +7,13 @@ def index
         @trading_names.map(&:registered_name).include? lookup_name.name
       end
     end
+
+    def destroy
+      trading_name = principal.firm.trading_names.registered.find(params[:id])
+      trading_name.destroy
```

Comment:
`destroy` always returns self even when the record has been removed from the db.